address-space-controller/src/main/java/io/enmasse/controller/ExportsController.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.controller;
+
+import io.enmasse.address.model.*;
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ExportsController implements Controller {
+    private static final Logger log = LoggerFactory.getLogger(ExportsController.class.getName());
+
+    private static final String KIND_SERVICE = "Service";
+    private static final String KIND_CONFIGMAP = "ConfigMap";
+    private static final String KIND_SECRET = "Secret";
+
+    private final KubernetesClient client;
+
+    public ExportsController(KubernetesClient client) {
+        this.client = client;
+    }
+
+    @Override
+    public AddressSpace reconcile(AddressSpace addressSpace) {
+        Map<String, List<ExportSpec>> exportsMap = new HashMap<>();
+        for (EndpointSpec endpointSpec : addressSpace.getSpec().getEndpoints()) {
+            if (endpointSpec.getExports() != null) {
+                exportsMap.put(endpointSpec.getName(), endpointSpec.getExports());
+            }
+        }
+
+        for (EndpointStatus endpointStatus : addressSpace.getStatus().getEndpointStatuses()) {
```

Comment:
The issue talked about the export including the CA, but I don't see that in the implementation.

Reply:
Fixed.